### PR TITLE
[#103] Update logic to get twitter username

### PR DESCRIPTION
**Pull request by JNicolao**
Make sure the checklist below is checked before PR merge, mark if not applicable

## PR Checklist

- [-] UI created matched design
- [-] For evey color class added, dark: is defined, images have alt and \_blank links have rel
- [-] API is integrated as required
- [-] Page is responsive

## What is the Purpose?
To update logic to get twitter username from both domains

Briefly describe what the PR addresses
[#103]

## What was the approach?
check if the URL contains either twitter.com or x.com. split the string to extract the username based on which domain is present.

## Are there any concerns to addressed further before or after merging this PR?
No

State some additional info if any. For instance running `install` or setting some environment variable(s)

## Mentions?
Mention persons you'd like to review this PR
@manuelgeek 

## Issue(s) affected?

List of issues addressed by this PR.
#103

### CODE REVIEW

File: components/sessions/SpeakersDetails.tsx [hunk 1]
```diff
@@ -14,6 +14,17 @@ export const SpeakersDetails = ({ session }: { session: Session }) => {
   if (!showChild) {
     return null
   }
+
+  const getTwitterUsername = (url: string) => {
```
Comment (danielotieno) [inline components/sessions/SpeakersDetails.tsx]: What does the function return? string, null or both?  `const getTwitterUsername = (url: string): string | null `

if it's not url, you can return null form the start 

`if (!url) return null;`

You can use regex to match both twitter and x URLs 

```
const match = url.match(/(?:twitter\.com|x\.com)\/([^/?#]+)/i);
return match?.[1] ?? null;
```

The complete function 

```js
const getTwitterUsername = (url: string): string | null => {
  if (!url) return null;
  
  const match = url.match(/(?:twitter\.com|x\.com)\/([^/?#]+)/i);
  return match?.[1] ?? null;
}

File: components/sessions/SpeakersDetails.tsx [hunk 2]
```diff
@@ -63,7 +74,7 @@ export const SpeakersDetails = ({ session }: { session: Session }) => {
                       >
                         @
                         {speaker.twitter
-                          ? speaker.twitter.split('twitter.com/')[1]
+                          ? getTwitterUsername(speaker.twitter)
```
Comment (danielotieno) [inline components/sessions/SpeakersDetails.tsx]: Here, why not? 
```js
{getTwitterUsername(speaker.twitter) ?? speaker.name}

**Review by danielotieno — CHANGES_REQUESTED** [review]

File: components/sessions/SpeakersDetails.tsx (hunk 1 above)
Comment (jnicolao) [reply components/sessions/SpeakersDetails.tsx]: This is fantastic @danielotieno . I don't know how I missed that. Exiting the loop early for non URLs will remove unnecessary processing. Let me make the changes.

File: components/sessions/SpeakersDetails.tsx (hunk 2 above)
Comment (jnicolao) [reply components/sessions/SpeakersDetails.tsx]: Thanks @danielotieno . Let me implement that. What do you think about
`{getTwitterUsername(speaker.twitter ?? '') ?? speaker.name}` to ensure that speaker.twitter is a string before passing it to `getTwitterUsername` ?

Comment (manuelgeek) [reply components/sessions/SpeakersDetails.tsx]: `{getTwitterUsername(speaker.twitter ?? '') || speaker.name}`
you can check the usage of || vs ??

Comment (danielotieno) [reply components/sessions/SpeakersDetails.tsx]: Or if you want to use ternary 

```js
{speaker.twitter 
  ? getTwitterUsername(speaker.twitter) || speaker.name
  : speaker.name}

Comment (jnicolao) [reply components/sessions/SpeakersDetails.tsx]: Yes ` || ` would be best in this case. I've already resolved this and pushed. Please review.

**Review by danielotieno — APPROVED** [review]

**Review by manuelgeek — APPROVED** [review]

**Pull request merged**